walk-in-interview/src/main/webapp/account/applicant/edit/script.js

```diff
@@ -0,0 +1,216 @@
+
+/**
+ * This file is specific to account/applicant/edit/index.html. 
+ * It renders the fields on the page dynamically.
+ */
+
+// TODO(issue/21): get the language from the browser
+const CurrentLocale = 'en';
+
+/**
+ * Import statements are static so its parameters cannot be dynamic.
+ * TODO(issue/22): figure out how to use dynamic imports
+ */
+import {AppStrings} from '../../../strings.en.js';
+import {API} from '../../../apis.js';
+
+import {APPLICANT_ID_PARAM, APPLICANT_DETAILS_PARAM, 
+  getRequirementsList, setErrorMessage} from '../../../common-functions.js';
+
+const HOMEPAGE_PATH = '../../../index.html';
+const STRINGS = AppStrings['applicant'];
+const BAD_REQUEST_STATUS_CODE = 400;
+
+window.onload = () => {
+  renderPageElements();
+};
+
+/**
+ * Adds text into the page.
+ */
+function renderPageElements() {
+  const applicantDetails = getApplicantDetails();
+
+  const backButton = document.getElementById('back');
+  backButton.innerText = STRINGS['back'];
+
+  const submitButton = document.getElementById('submit');
+  submitButton.setAttribute('value', STRINGS['submit']);
+  submitButton.setAttribute('type', 'submit');
+
+
+  const nameLabel = document.getElementById('name-label');
+  nameLabel.innerText = STRINGS['name'];
+
+  const name = document.getElementById('name');
+  name.setAttribute('type',  'text');
+  name.setAttribute('value', applicantDetails.name);
+
+  const skillsTitle = document.getElementById('skills-title');
+  skillsTitle.innerText = STRINGS['skills-title'];
+
+  renderSkillsList(applicantDetails.skills);
+}
+
+/**
+ * Gets the applicantId from the url.
+ * @return {String} The applicantId.
+ */
+function getApplicantId() {
+  const queryString = window.location.search;
+
+  const urlParams = new URLSearchParams(queryString);
+  if (urlParams === '') {
+    throw new Error('url params should not be empty');
+  }
+
+  return urlParams.get(APPLICANT_ID_PARAM);
+}
+
+/**
+ * Gets the applicant details from the url.
+ * @return {JSON} The applicant detail json.
+ */
+function getApplicantDetails() {
+  const queryString = window.location.search;
+
+  const urlParams = new URLSearchParams(queryString);
+  if (urlParams === '') {
+    throw new Error('url params should not be empty');
+  }
+
+  return JSON.parse(urlParams.get(APPLICANT_DETAILS_PARAM));
```

Comment:
how can url param become detail?

Reply:
we put the full applicant detail json in the url